test/e2e/nodepool_autorepair_test.go

```diff
@@ -0,0 +1,150 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func testNodePoolAutoRepair(mgmtClient crclient.Client, guestCluster *hyperv1.HostedCluster, guestClient crclient.Client, clusterOpts core.CreateOptions) func(t *testing.T) {
+	return func(t *testing.T) {
+		t.Parallel()
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
the context needs to be the shared all around so all functions are signalled together.
See other tests
```
ctx, cancel := context.WithCancel(testContext)
defer cancel()
```

Reply:
I cannot share the context with the tests under `nodepool_test.go`. I've tested it and when I set some tests in parallel, the context gets cancelled before the test ends. This is why I've set a concrete context per test

Reply:
mm so we need to understand why. All tests need to share context as they do now, so all running routines are closed when test are cancelled/timeout.

Reply:
The reason it's that I have a `defer cancel()` in the TestNodePool function, which is the one who calls the other tests (like autorepair).

I will remove the `defer cancel()` from this function and progress the cancel to the nexts inner tests, having there the `defer cancel()` and cancelling the context on failure.